onedal/linear_model/linear_model.py

```diff
@@ -293,15 +293,15 @@ def fit(self, X, y, queue=None):
         """
         module = self._get_backend("linear_model", "regression")
 
-        X = _check_array(
-            X,
-            dtype=[np.float64, np.float32],
-            force_all_finite=False,
-            ensure_2d=False,
-            copy=self.copy_X,
-        )
+        if not isinstance(X, np.ndarray):
+            X = np.asarray(X)
+
+        dtype = get_dtype(X)
+        if dtype not in [np.float32, np.float64]:
+            dtype = np.float64
+            X = X.astype(dtype, copy=self.copy_X)
 
-        y = np.asarray(y).astype(dtype=get_dtype(X))
+        y = np.asarray(y).astype(dtype=dtype)
```

Comment:
This is still unclear to me why we have this perf drop here. How much this impl differ from daal4py CPU branch?